softwerkskammer/lib/persistence/sqlitePersistence.js

```diff
@@ -0,0 +1,179 @@
+const conf = require("simple-configure");
+const Database = require("better-sqlite3");
+const path = require("node:path");
+const R = require("ramda");
+const loggers = require("winston").loggers;
+
+const sqlitedb = conf.get("sqlitedb");
+const db = new Database(path.join(__dirname, sqlitedb));
+const scriptLogger = loggers.get("scripts");
+scriptLogger.info(`DB = ${sqlitedb}`);
+const CONFLICTING_VERSIONS = conf.get("beans").get("constants").CONFLICTING_VERSIONS;
+
+function escape(str = "") {
+  if (typeof str === "string") {
+    return `'${str.replaceAll("'", "''")}'`;
+  }
+  if (typeof str === "number") {
+    return str;
+  } else {
+    return str + "";
+  }
+}
+
+function asSqliteString(obj) {
+  return `${escape(JSON.stringify(obj))}`;
+}
+
+function execWithTry(command) {
+  try {
+    db.exec(command);
+  } catch (e) {
+    const errortext = e.toString();
+    if (!(errortext.startsWith("SqliteError: index") && errortext.endsWith("already exists"))) {
+      // we expect re-entrance of index creation
+      // eslint-disable-next-line no-console
+      console.error(errortext);
+    }
+  }
+}
+
+module.exports = function sqlitePersistenceFunc(collectionName, extraColumns) {
+  const extraCols = extraColumns ? extraColumns.split(",") : [];
+  function create() {
```

Comment:
Why attempting to create the database tables every time the beans are instantiated, instead of creating the tables only once during migration?

Reply:
relict from database tests. see below "recreateForTest". Will change that after successful merge

Reply:
additionally: migration needs this to create the tables and indices